submitit/core/job_environment.py

```diff
@@ -58,6 +58,14 @@ def activated(self) -> bool:
     def hostname(self) -> str:
         return socket.gethostname()
 
+    @property
+    def hostnames(self) -> List[str]:
+        env_key = "nodes"
+        if env_key not in self._env:
+            raise RuntimeError("Hostnames not available for the {self.cluster} cluster")
+        node_list = os.environ.get(self._env[env_key], "")
+        return self._parse_nodelist(node_list)
```

Comment:
That seems overly complex in my opinion (the fact that everything else is key-based is mostly historical.
It seems simpler to have only 
```python
   @property
    def hostnames(self) -> List[str]:
        raise RuntimeError("Hostnames not available for the {self.cluster} cluster")
```
Then you can override it in the Slurm environment. That would remove one layer

Reply:
Agreed. I mostly tried to comply with the patterns used elsewhere in this code base, but this is not necessarily a great idea.